src/core/memory.cc

```diff
@@ -1548,9 +1548,9 @@ void configure(std::vector<resource::memory> m, bool mbind,
 
             if (r == -1) {
                 char err[1000] = {};
-                strerror_r(errno, err, sizeof(err));
+                char *msg = strerror_r(errno, err, sizeof(err));
```

Comment:
Wow, I didn't remember this, but it turns out there are actually two implementations of strerror_r(), and this one (which returns a char* which the caller should use) is actually the non-standard GNU-specific one - but apparently, the one we're indeed using normally.

How did you find this problem? Did you see a wrong error message?

Reply:
@nyh I found this by building seastar with `-Wunused-result` and `-Werror=unused-result` on gcc11.

To enable building seastar with those flags, I had to apply an additional [patch](https://gist.github.com/crackcomm/5b84f1ff6950bdefa8d8cc2db18ea0f9). However, I noticed that some of the results were intentionally unused, so I decided not to submit a PR.

Reply:
@nyh now that I think about, maybe it's not a bad idea to apply the above patch? It explicitly states that we're aware those results are supposed to be used and we just ignore them. It makes it easier for reader to infer that this is what's happening.